tests/modifier-tests.lisp

```diff
@@ -59,3 +59,67 @@
       (is (cflonum= 1 (qvm::nth-amplitude q 0)))
       (loop :for i :from 1 :below (expt 2 3) :do
         (is (cflonum= 0 (qvm::nth-amplitude q i)))))))
+
+(deftest test-forked-rx ()
+  (with-execution-modes (:compile :interpret)
+    (let* ((p-first-branch
+             (with-output-to-quil "FORKED RX(0, pi) 0 1"))
+           (p-second-branch
+             (with-output-to-quil
+               "X 0"
+               "FORKED RX(0, pi) 0 1")))
+      ;; Since q0 should be in the zero state, the FORKED should
+      ;; produce RX(0) 1, hence leaving q1 in the zero state.
+      (is (cflonum= 0 (qvm:probability (qvm::nth-amplitude (run-program 2 p-first-branch) 3))))
+      ;; q1 is now in the one state, forked should produce RX(pi) 1,
+      ;; leaving q1 in the one state.
+      (is (cflonum= 1 (qvm:probability (qvm::nth-amplitude (run-program 2 p-second-branch) 3)))))))
+
+(deftest test-forked-rx-with-controlled-and-dagger ()
+  (with-execution-modes (:compile :interpret)
+    (let* ((p-first-branch
+             (with-output-to-quil "CONTROLLED DAGGER FORKED RX(0, pi) 0 1 2"))
+           (p-second-branch
+             (with-output-to-quil
+               "X 0"
+               "X 1"
+               "CONTROLLED DAGGER FORKED RX(0, pi) 0 1 2")))
+      ;; Since q0, q1 should be in the zero state, the FORKED should
+      ;; produce RX(0) 2, hence leaving q2 in the zero state.
+      (is (cflonum= 0 (qvm:probability (qvm::nth-amplitude (run-program 3 p-first-branch) 7))))
+      ;; q0, q1 are now in the one state, forked should produce RX(pi) 2,
+      ;; leaving q2 in the one state.
+      (is (cflonum= 1 (qvm:probability (qvm::nth-amplitude (run-program 3 p-second-branch) 7)))))))
+
+(deftest test-multiply-forked-rx ()
+  ;; AKA the uniformly controlled rotation
+  ;; TODO Cover all possibilities.
+  (with-execution-modes (:compile :interpret)
+    (let* ((p-first-branch
+             (with-output-to-quil "FORKED FORKED RX(0, 0, 0, pi) 0 1 2"))
+           (p-second-branch
+             (with-output-to-quil
+               "X 0"
+               "X 1"
+               "FORKED FORKED RX(0, 0, 0, pi) 0 1 2")))
+      ;; Since q0, q1 should be in the zero state, the FORKED should
+      ;; produce RX(0) 2, hence leaving q2 in the zero state.
+      (is (every (alexandria:curry #'cflonum= 0)
+                 (map-amplitudes (run-program 3 p-first-branch) #'qvm:probability)))
```

Comment:
`map-amplitudes` doesn't return anything. You might just peek at the amplitudes directly. Also, it's not possible for every amplitude to be 0.